### PR TITLE
Fit model

### PR DESCRIPTION
**Pull request by raodz**
## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests update
- [ ] Documentation update
- [ ] Changes that do not directly affect the source code or tests

## Checklist
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/prepare_datasets.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import os
+import pandas as pd
+from src.data import BaldDataset
+from src.config_class import BaldOrNotConfig
+
```
Comment (skrzypczykt) [inline scripts/prepare_datasets.py:8]: isort, if name main

File: scripts/prepare_datasets.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+labels_df = pd.read_csv(labels_ds_path)
+
+cleaned_df = BaldDataset.get_cleaned_df(labels_df, images_dir)
+merged_df = BaldDataset.prepare_merged_dataframe(subsets_df, labels_df)
+converted_df = BaldDataset.convert_image_id_column_to_float(merged_df)
+balanced_df = BaldDataset.balance_classes(
+    converted_df,
+    X_cols=["image_id", "partition"],
+    y_col="labels",
+    minor_class_multiplier=config.training_params.minor_class_multiplier,
```
Comment (skrzypczykt) [inline scripts/prepare_datasets.py]: rename to minority

File: src/data.py [hunk 3]
```diff
@@ -311,3 +314,82 @@ def create_subset_dfs(  # problem
...
+            df[image_id_col].str.replace(".jpg", "", regex=False).astype(float)
+        )
+        return df
+
+    @staticmethod
+    def balance_classes(
+        df: pd.DataFrame,
+        X_cols: list,
+        y_col: str,
+        minor_class_multiplier: int = 10,
```
Comment (skrzypczykt) [inline src/data.py]: no default

File: scripts/prepare_datasets.py (hunk 2 above)
Comment (skrzypczykt) [reply scripts/prepare_datasets.py]: move out from offline preprocessing to online one

File: src/data.py [hunk 4]
```diff
@@ -311,3 +314,82 @@ def create_subset_dfs(  # problem
...
+        target_minority_size = minority_count * minor_class_multiplier
+
+        # Set the oversampling strategy
+        sampling_strategy = {minority_class: target_minority_size}
+
+        # Oversampling the minority class using SMOTE
+        sm = SMOTE(sampling_strategy=sampling_strategy, random_state=42)
+        X_resampled, y_resampled = sm.fit_resample(X, y)
+
+        # Undersampling the majority class
```
Comment (skrzypczykt) [inline src/data.py]: nie

**Comment by skrzypczykt** [comment]
https://www.tensorflow.org/tutorials/structured_data/imbalanced_data

File: scripts/prepare_datasets.py (hunk 1 above)
Comment (raodz) [reply scripts/prepare_datasets.py:8]: ok

File: src/data.py (hunk 3 above)
Comment (raodz) [reply src/data.py]: ok

File: src/data.py (hunk 4 above)
Comment (raodz) [reply src/data.py]: Ok

**Pull request closed without merging**